### DISCUSSION TITLE
JS SDK client.users.refresh() returns 401

### DISCUSSION THREAD

**Discussion by IzMichael** [pocketbase/pocketbase#568] [Q&A] [answered]
Hiya,

Whenever I try to refresh the auth using the [JS SDK](https://github.com/pocketbase/js-sdk), it fails and returns the following 401 error message: `401: The request requires valid user authorization token to be set.`.
```javascript
if ($user.token) $user = await client.users.refresh();
```
```json
{"code":401,"message":"The request requires valid user authorization token to be set.","data":{}}
```
From taking a look at the network tab in DevTools, the request is sending the access token as an admin token, instead of a user token.
![image](https://user-images.githubusercontent.com/57115702/191219641-436aeacc-1660-4d02-9017-989e447bf156.png)
Is there any way for me to fix this, or do I just have to use the REST API instead?

**Comment by ganigeorgiev** [comment, marked as answer]
If the Authorization header is set with an Admin token, then that means that the `client.authStore` has Admin auth model and token (admins and users are 2 separate entities in PocketBase).

`client.users.refresh()` refresh the token and auth data only for the currently authenticated user models.

To refresh the data of the currently authenticated admin, you can call `client.admins.refresh()` (corresponds to the admin API endpoint - `POST /api/admins/refresh`).

**Reply by IzMichael** [reply]
I'm trying to refresh a user, sorry if I wasn't clear in my question.
I'm logging in as a user, and then trying to refresh that user.
```javascript
$user = await client.users.authViaEmail(email, password);
if ($user.token) $user = await client.users.refresh();
```
Network tab confirms both requests are going to `/api/users/auth-via-email` and `/api/users/refresh`

**Reply by IzMichael** [reply]
I'll note that I don't have any issues when I send the POST to `/api/users/refresh` manually via fetch as a user token, instead of the sdk

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce the issue. The Authorization header is correctly using the `User` schema for me when doing:

```js
await client.users.authViaEmail("test@example.com", "123456");
await client.users.refresh();
```

If you have an admin auth call before the `users.refresh` then it is possible that the authStore has changed with the admin data, aka. the following will not work:

```js
await client.users.authViaEmail("user@example.com", "1234567890");
await client.admins.authViaEmail("admin@example.com", "1234567890");
await client.users.refresh()
```

The AuthStore works only with 1 auth model at a time.

**Reply by IzMichael** [reply]
I think I've found the issue, I had been setting the localStorage value to the object (using a svelte binding), at the same time that the authStore was doing it. There must have been some conflict in the versions saved or something.

Thanks for the assistance!
